Configurable bibliography file generation detection

ahmadsalim opened the pull request:
This PR fixes #8 by making the bibliography file ending configurable (e.g. allowing changing '.bbl' which is used by BibTeX to '.bcf' which is used by biber), and the string that is to detect when to run the specified flavor/engine.

The commit is partially inspired by answers found in:
http://tex.stackexchange.com/questions/83715/biber-backend-and-vim-latex.

Any comments are welcome.

ahmadsalim commented:
Actually, I discovered that the file extension was not used for anything contrary to what the TeX.SE answers suggested, and so I have updated the PR to only have a customizable bibliography file generation detection string.
This string was hardcoded to be '\\bibdata', but that only works for BibTex and not tools like biber which use things starting with '\\abx', e.g. '\\abx@aux@cite'.

tyll commented:
Can you please squash the commits together so only commits for valid changes remain?

ahmadsalim commented:
@tyll Yeah, sure will do.

ahmadsalim commented:
OK, the commits are now squashed into one and the commit message should be more sensible now.

tyll commented:
Ok, now it is more obvious. I am not sure whether it really needs to be configurable, since `Tex_IsPresentInFile` accepts a regex. Therefore it would be possible to use something like `\\bibdata\|\\abx` , then it would support biber and bibtex without the need to change the configuration. Can you please test this and tell me your opinion? If you agree and adjust the patch, can you please also update the comment in `compiler.vim` to mention that `\\abx` is needed for biber?

ahmadsalim commented:
Hmm, so I thought that it might be more future-proof if it was configurable, but if that is not really needed I can remove it. 
I could probably also change the configurable setting so that it is `'\\bibdata|\\abx'` I guess, which would make it work automatically for both engines while not requiring user-intervention by default.
If you prefer any of those I will do that, otherwise I will try to test things and update the comments as you suggest.

tyll commented:
IMHO it making it configurable just makes it more complicated. We can also update the default regex in the future. So please just update the comment/code directly and test it. Thank you.

ahmadsalim commented:
OK, I have done as suggested.
It seems to work with both the `biber` and `BibTeX` backends.

I tested it as follows:
* Removed .aux, .bbl and .bcf files
* Changed `g:Tex_BibTeXFlavor` to preferred backend
* Ran Vim-LaTeX compilation process on a file with BibLaTeX bibliography and confirmed that LaTeX-backend-LaTeX were run and that the backend stated was the correct one
* Checked outputs

Hopefully, everything should be alright now.

tyll commented:
ok, can you please squash the commits, then I'll merge it.

ahmadsalim commented:
OK, I have squashed the commits again.

tyll commented:
Thank you very much

ahmadsalim commented:
Sure, thanks for merging.
